locus/src/main/java/com/birjuvachhani/locus/LocationProvider.kt

```diff
@@ -83,7 +83,7 @@ internal class LocationProvider(context: Context) {
         fun startUpdates() {
             val callback = object : LocationCallback() {
                 override fun onLocationResult(result: LocationResult) {
-                    onUpdate(LocusResult.success(result.lastLocation))
+                    result.lastLocation?.let { LocusResult.success(it) }?.let { onUpdate(it) }
```

Comment:
Why are you using `lastLocation?` as if it is nullable? Android studio warns about it being non-null?

<img width="1128" alt="image" src="https://github.com/BirjuVachhani/locus-android/assets/20423471/69ec2a08-3a26-4229-aac0-ef0138c99280">

Also it can be simplify like this:

```kotlin
result.lastLocation?.let { onUpdate(LocusResult.success(it)) }
```

Reply:
In the new version of Google Play Service Location, the `lastLocation` is nullable, and also I have simplified it with a new commit.